include/swift/ABI/Metadata.h

```diff
@@ -1675,6 +1676,19 @@ struct TargetFunctionTypeMetadata : public TargetMetadata<Runtime> {
     return reinterpret_cast<const uint32_t *>(getParameters() +
                                               getNumParameters());
   }
+
+  TargetFunctionMetadataDifferentiabilityKind<StoredSize>
+  getDifferentiabilityKind() const {
+    if (isDifferentiable()) {
+      auto addr = llvm::alignAddr(
+          getParameterFlags() + getNumParameters(),
+          llvm::Align(alignof(typename Runtime::StoredPointer)));
+      return *reinterpret_cast<
+          TargetFunctionMetadataDifferentiabilityKind<StoredSize> *>(addr);
+    }
+    return TargetFunctionMetadataDifferentiabilityKind<StoredSize>
+        ::NonDifferentiable;
+  }
```

Comment:
You could use `swift/ABI/TrailingObjects.h` to help with the trailing allocation management here. Where do you initialize the tail storage?

Reply:
Yeah, initially I wanted to use that but noticed that parameters and parameter flags are manually, so I followed the same approach. Would it be odd to mix both, no? The allocation size calculation is in `FunctionCacheEntry::getExtraAllocationSize()`; I aligned the previously calculated size (because there are a few 32-bit ints for parameter flags) and appended a differentiability word after it.